### PR TITLE
update syncval code generation for synchronization2

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
This is depenedent on https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2554

### CODE REVIEW

**Comment by ncesario-lunarg** [comment]
Looks like the windows appveyor errors might be valid @jeremyg-lunarg?

**Comment by jeremyg-lunarg** [comment]
> Looks like the windows appveyor errors might be valid @jeremyg-lunarg?

Yep the failures in this PR are real.

File: layers/sync_utils.h [hunk 1]
```diff
@@ -21,31 +21,38 @@
...
+VkAccessFlags2KHR CompatibleAccessMask(VkPipelineStageFlags2KHR stage_mask);
 
-VkPipelineStageFlags WithLaterPipelineStages(VkPipelineStageFlags stage_mask);
+VkPipelineStageFlags2KHR WithEarlierPipelineStages(VkPipelineStageFlags2KHR stage_mask);
 
-int GetGraphicsPipelineStageLogicalOrdinal(VkPipelineStageFlags flag);
+VkPipelineStageFlags2KHR WithLaterPipelineStages(VkPipelineStageFlags2KHR stage_mask);
 
-VkPipelineStageFlags GetLogicallyEarliestGraphicsPipelineStage(VkPipelineStageFlags inflags);
+int GetGraphicsPipelineStageLogicalOrdinal(VkPipelineStageFlags2KHR flag);
```
Comment (jzulauf-lunarg) [inline layers/sync_utils.h:47]: unsigned or uint32t?

File: layers/synchronization_validation.cpp [hunk 2]
```diff
@@ -2969,8 +2970,7 @@ VkPipelineStageFlags ResourceAccessState::GetOrderedStages(const OrderingBarrier
 void ResourceAccessState::UpdateFirst(const ResourceUsageTag &tag, SyncStageAccessIndex usage_index, SyncOrdering ordering_rule) {
     // Only record until we record a write.
     if (first_accesses_.empty() || IsRead(first_accesses_.back().usage_index)) {
-        const VkPipelineStageFlags usage_stage =
-            IsRead(usage_index) ? static_cast<VkPipelineStageFlags>(PipelineStageBit(usage_index)) : 0U;
+        const VkPipelineStageFlags2KHR usage_stage = IsRead(usage_index) ? PipelineStageBit(usage_index) : 0U;
```
Comment (jzulauf-lunarg) [inline layers/synchronization_validation.cpp:2973]: :)

File: scripts/sync_val_gen.py [hunk 3]
```diff
@@ -29,107 +29,118 @@
...
 
 # Some DRY constants
-host_stage = 'VK_PIPELINE_STAGE_HOST_BIT'
-top_stage ='VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT'
-bot_stage ='VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT'
+host_stage = 'VK_PIPELINE_STAGE_2_HOST_BIT_KHR'
+top_stage ='VK_PIPELINE_STAGE_2_TOP_OF_PIPE_BIT_KHR'
+bot_stage ='VK_PIPELINE_STAGE_2_BOTTOM_OF_PIPE_BIT_KHR'
 
 # Snipped from chapters/synchronization.txt -- tag v1.2.169
```
Comment (jzulauf-lunarg) [inline scripts/sync_val_gen.py:36]: Attribution version is no earlier than 170, right?

**Review by jzulauf-lunarg — COMMENTED** [review]
The only nit that matters is the excerpt version string update.  Updating is unwieldy enough with lying to the poor, next maintainer.

File: layers/sync_utils.h (hunk 1 above)
Comment (jeremyg-lunarg) [reply layers/sync_utils.h:47]: this returns -1 in an error case and at this point I think we need to let it lie for now.

File: scripts/sync_val_gen.py (hunk 3 above)
Comment (jeremyg-lunarg) [reply scripts/sync_val_gen.py:36]: fixed.  generated code didn't change.

File: layers/sync_utils.cpp [hunk 4]
```diff
@@ -71,17 +72,55 @@ VkPipelineStageFlags ExpandPipelineStages(VkPipelineStageFlags stage_mask, VkQue
...
         expanded &= ~VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT;
-        expanded |= syncAllCommandStagesByQueueFlags.at(VK_QUEUE_GRAPHICS_BIT) & ~disabled_feature_mask;
-        expanded &= ~VK_PIPELINE_STAGE_HOST_BIT;
+        // Make sure we don't pull in the HOST stage from expansion, but keep it if set by the caller.
+        // The syncAllCommandStagesByQueueFlags table includes HOST for all queue types since it is
+        // allowed but it shouldn't be part of ALL_GRAPHICS
+        expanded |=
+            syncAllCommandStagesByQueueFlags.at(VK_QUEUE_GRAPHICS_BIT) & ~disabled_feature_mask & ~VK_PIPELINE_STAGE_HOST_BIT;
+    }
+    if (VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR & stage_mask) {
```
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply layers/sync_utils.cpp:81]: we need to roll this into the spec issue for getting the stage info

**Pull request merged**